accounting/src/delta/delta_data_collection/tests/delta_delta_undo_tests.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2021-2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+
+use rstest::rstest;
+
+type ThreeCollections = (
+    DeltaDataCollection<i32, char>,
+    DeltaDataCollection<i32, char>,
+    DeltaDataCollection<i32, char>,
+);
+
+#[rstest]
+#[rustfmt::skip]
+#[case(DataDelta::new(None,      Some('a')), DataDelta::new(Some('a'), None))]
+#[case(DataDelta::new(None,      Some('a')), DataDelta::new(Some('a'), Some('b')))]
+#[case(DataDelta::new(Some('a'), Some('b')), DataDelta::new(Some('b'), Some('c')))]
+#[case(DataDelta::new(Some('a'), Some('b')), DataDelta::new(Some('b'), None))]
+#[case(DataDelta::new(Some('a'), None),      DataDelta::new(None,      Some('b')))]
+fn delta_delta_undo_associativity(
+    #[case] delta1: DataDelta<char>,
+    #[case] delta2: DataDelta<char>,
+) {
+    {
+        // (Delta1 + Delta2) + Undo = Delta1
+        // every delta goes into separate collection
+        let (mut collection1, collection2, collection3) =
+            make_collections_with_undo(delta1.clone(), delta2.clone());
+        let expected_collection = collection1.clone();
```

Comment:
If both are expected to have the same result, please take this out of the scope.